plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java

```diff
@@ -4941,11 +4941,16 @@ public void testProjectionPushdownReadsLessData()
                 getSession(),
                 selectQuery,
                 statsWithPushdown -> {
+                    DataSize physicalDataSizeWithPushdown = statsWithPushdown.getPhysicalInputDataSize();
                     DataSize processedDataSizeWithPushdown = statsWithPushdown.getProcessedInputDataSize();
                     assertQueryStats(
                             sessionWithoutPushdown,
                             selectQuery,
-                            statsWithoutPushdown -> assertThat(statsWithoutPushdown.getProcessedInputDataSize()).isGreaterThan(processedDataSizeWithPushdown),
+                            statsWithoutPushdown -> {
+                                //TODO (https://github.com/trinodb/trino/issues/17156) add dereference pushdown on the physical layer
```

Comment:
We may need more test data to show data pruning happens on read and/or disable Parquet compression (on write).
3000 'Z' characters will compress very well, and the coalescing adjacent reads should kick in